Tests/Regex/DPLRegularExpressionSpec.m

```diff
@@ -33,7 +33,31 @@
                                                         @"this": @"thisvalue",
                                                         @"that": @"thatvalue" });
     });
-    
+
+    it(@"should match an array", ^{
+        DPLRegularExpression *expression = [DPLRegularExpression regularExpressionWithPattern:@"/hello/:list"];
+
+        DPLMatchResult *matchResult = [expression matchResultForString:@"/hello/list[]=item1&list[]=item2"];
+        expect(matchResult.match).to.beTruthy();
+        expect(matchResult.namedProperties).to.equal(@{ @"list": @[ @"item1", @"item2" ] });
```

Comment:
I believe the expected behavior is as follows: 
(Note: https://www.w3.org/Addressing/URL/url-spec.txt)

```
Path params:
@"/hello/list[]=item1&list[]=item2      ====>    @{ @"list": @"list[]=item1&list[]=item2" }

Query params:
@"/hello/?list[]=item1&list[]=item2    ====>    @{ @"list": @[ @"item1", @"item2" ] }
```